PyDynamic/uncertainty/interpolate.py

```diff
@@ -303,6 +303,84 @@ def interp1d_unc(
                     (t_new[interp_range] - t_hi) ** 2 * uy_prev_sqr
                     + (t_new[interp_range] - t_lo) ** 2 * uy_next_sqr
                 ) / (t_hi - t_lo)
+    elif kind == "cubic":
+        # Calculate boolean arrays of indices from t_new which are outside t's bounds...
+        extrap_range_below = t_new < np.min(t)
+        extrap_range_above = t_new > np.max(t)
+        extrap_range = extrap_range_below | extrap_range_above
+        # .. and inside t's bounds.
+        interp_range = ~extrap_range
+
+        # Initialize the result array for the standard uncertainties.
+        uy_new = np.empty_like(y_new)
+
+        # Initialize the sensitivity matrix of shape (M, N)
+        C = np.zeros((len(t_new), len(uy)), "float64")
+
+        # First extrapolate the according values if required and then
+        # compute interpolated uncertainties following White, 2017.
+
+        # If extrapolation is needed, fill in the values provided via fill_unc.
+        if np.any(extrap_range):
+            # At this point fill_unc is either a float (np.nan is a float as well) or
+            # a 2-tuple of floats. In case we have one float we set uy_new to this value
+            # inside the extrapolation range.
+            if isinstance(fill_unc, float):
+                uy_new[extrap_range] = fill_unc
+            else:
+                # Now fill_unc should be a 2-tuple, which we can fill into uy_new.
+                uy_new[extrap_range_below], uy_new[extrap_range_above] = fill_unc
+
+            if returnC:
+                # In each row of C corresponding to an extrapolation value below the
+                # original range set the first column to 1 and in each row of C
+                # corresponding to an extrapolation value above the original range set
+                # the last column to 1. It is important to do this before
+                # interpolating, because in general those two columns can contain
+                # non-zero values in the interpolation range.
+                C[:, 0], C[:, -1] = extrap_range_below, extrap_range_above
+
+        # If interpolation is needed, compute uncertainties following White, 2017.
+        if np.any(interp_range):
```

Comment:
I already thought about that during the implementation and was undecided. Because of that, I opted for the version, that keeps a very clear separation of the spline-functionality - at least for your first review. 

I will refactor that out 👍